pkg/command/lint/command.go

```diff
@@ -81,13 +82,23 @@ func Command() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			if verbose {
+			if verbose || errorOnInvalidResource {
+				hasInvalidObjects := false
+				prefix := "Warning"
+				if errorOnInvalidResource {
+					prefix = "Error"
+				}
 				for _, lintCtx := range lintCtxs {
 					for _, invalidObj := range lintCtx.InvalidObjects() {
-						fmt.Fprintf(os.Stderr, "Warning: failed to load object from %s: %v\n", invalidObj.Metadata.FilePath, invalidObj.LoadErr)
+						hasInvalidObjects = true
+						_, _ = fmt.Fprintf(os.Stderr, "%s: failed to load object from %s: %v\n", prefix, invalidObj.Metadata.FilePath, invalidObj.LoadErr)
 					}
 				}
+				if hasInvalidObjects && errorOnInvalidResource {
+					return fmt.Errorf("unable to load all objects")
```

Comment:
I'm not sure if we should short circuit here. How about you have 10 files but only one is invalid. Instead of getting lint errors for all of them you'll get only information about invalid one. I think better if we add this results to other linters so we can have them in json or sarif format as well helping automation tools to mark them as problematic. 
What do you think about this? On the other hand we can start small (merge this PR) and then iterate on what we have.

Reply:
I was thinking about this a well. These errors also aren't "pretty printed". I am not sure how to pass the results down. I would need to either append them to the `result` array at end. I also would need to fill in the `Result` object. 

Happy to do it but it was a larger change. Any impl suggestions?

Reply:
@janisz, don't mean to be pushy but any thoughts here? I can definitely "hack" it into a result struct but want to make sure we agree before I do it.

Reply:
I'm sorry I was on holiday with limited access to internet. I'll try to find a way how to simplify the implementation